agent/engine/task_manager.go

```diff
@@ -254,7 +254,7 @@ func (mtask *managedTask) handleContainerChange(containerChange dockerContainerC
 			}
 			// If docker returned a transient error while trying to stop a container,
 			// reset the known status to the current status and return
-			cannotStopContainerError, ok := event.Error.(*CannotStopContainerError)
+			cannotStopContainerError, ok := event.Error.(CannotStopContainerError)
```

Comment:
While this fixes it for now, this doesn't prevent the same problem from happening again.  I would much prefer if we defined an interface for this and type-asserted to the interface.  Interface type-assertions don't matter whether you have a struct or a pointer since either can satisfy the interface.  See [here](https://play.golang.org/p/T07I-lGfiP).

Something like this:

```go
type maybeUnretriableError interface {
	IsUnretriableError() bool
}
```

You should be able to just define this above `handleContainerChange` and use it here.